### PR TITLE
feat: add support for value transfer

### PR DESCRIPTION
**Pull request by smartcontracts**
Adds an extra parameter for non-zero value transfer.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
-[submodule "/Users/james/devel/ExcessivelySafeCall/lib/ds-test"]
-	path = /Users/james/devel/ExcessivelySafeCall/lib/ds-test
+[submodule "lib/ds-test"]
```
Comment (smartcontracts) [inline .gitmodules:1]: Had to update this to get submodule cloning to work

Comment (smartcontracts) [reply .gitmodules:1]: Got updated to latest commit too, can revert that if you're opinionated about it

**Comment by prestwich** [comment]
Fixed the submodule thing. Please rebase onto latest main 😅

**Comment by smartcontracts** [comment]
> Fixed the submodule thing. Please rebase onto latest main 😅

Rebased. Updated the submodule name but can keep the existing name with your path if you want. Up to you!

File: src/test/ExcessivelySafeCall.t.sol [hunk 2]
```diff
@@ -46,11 +48,22 @@ contract ContractTest is DSTest {
...
+            0,
             abi.encodeWithSelector(CallTarget.any.selector, 5)
         );
         assertTrue(_success);
         assertEq(_ret.length, 0);
         assertEq(t.called(), 5);
+
+        (_success, _ret) = target.excessivelySafeCall(
+            100_000,
+            69,
```
Comment (prestwich) [inline src/test/ExcessivelySafeCall.t.sol:60]: nice

**Review by prestwich — APPROVED** [review]

File: src/test/ExcessivelySafeCall.t.sol (hunk 2 above)
Comment (smartcontracts) [reply src/test/ExcessivelySafeCall.t.sol:60]: niceee

**Pull request merged**